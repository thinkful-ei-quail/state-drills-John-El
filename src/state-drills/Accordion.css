.Accordion{
    display: grid;
    list-style-type: none;
    place-items: center;
    margin-left: -1.4em;
}

.AccButt{
    width: 40vw;
    height: 3vh;
    font-size: 2vw;
    letter-spacing: .2em;

}

.Content-p{
    font-size: 4vw;
    font-weight: 90;
    text-align: start;
}
