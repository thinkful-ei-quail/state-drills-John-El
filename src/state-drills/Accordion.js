import './Accordion.css'
import React, { Component } from 'react'

export default class Accordion extends Component {
    static defaultProps = {
        sections: [
            {
                title: 'A String Containg the Title',
                content: 'A String Containing the Content'
            }]
    };

    state = { activeSelectionIndex: null }

    renderSection(section, dex, activeSelectionIndex) {
        return (
            <li className='Accordion-li'>
                <button className='AccButt' onClick={() => this.setState({ activeSelectionIndex: dex })} >
                {section.title}
                </button>

                {(activeSelectionIndex === dex) && <p className='Content-P'>{section.content}</p>}
            </li>
        )
    }


    render() {
        const { activeSelectionIndex } = this.state
        const { sections } = this.props
        return (
            <ul className='Accordion'>
                {sections.map((section, dex) =>
                    this.renderSection(section, dex, activeSelectionIndex)
                )}
            </ul>
        )
    }

}
