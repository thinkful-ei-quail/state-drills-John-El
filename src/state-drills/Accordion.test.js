
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer'
import Accordion from './Accordion'


const sectionsProp=[
    {
        title: 'Section 1',
        content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
      },
      {
        title: 'Section 2',
        content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
      },
      {
        title: 'Section 3',
        content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
      }
]


describe('Accordion Comp.', ()=>{
    it('renders an empty li when the sections prop is not supplied',()=>{
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})


describe('Accordion Comp.', ()=>{
    it('renders no sections as active by default',()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

describe('Accordion Comp.', ()=>{
    it('opens a clicked section',()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

describe('Accordion Comp.', ()=>{
    it('only opens the last section when multiple sections have been clicked',()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
