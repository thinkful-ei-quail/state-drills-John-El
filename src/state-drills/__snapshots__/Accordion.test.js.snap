// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Comp. only opens the last section when multiple sections have been clicked 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p
      className="Content-P"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Comp. opens a clicked section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      className="Content-P"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Comp. renders an empty li when the sections prop is not supplied 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      A String Containg the Title
    </button>
  </li>
</ul>
`;

exports[`Accordion Comp. renders no sections as active by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion-li"
  >
    <button
      className="AccButt"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
